api/core/v1alpha1/etcd.go

```diff
@@ -328,6 +328,10 @@ type EtcdSpec struct {
 	// +optional
 	// +kubebuilder:validation:XValidation:message="etcd.spec.VolumeClaimTemplate is an immutable field",rule="self == oldSelf"
 	VolumeClaimTemplate *string `json:"volumeClaimTemplate,omitempty"`
+	// RunAsRoot defines whether the securityContext of the pod specification should indicate that the containers shall
+	// run as root. By default, they run as non-root with user 'nobody'.
+	// +optional
+	RunAsRoot *bool `json:"runAsRoot,omitempty"`
```

Comment:
Many of the pod/container specific configuration is part of `Etcd.Spec.Etcd` (the naming sucks right now - this is the baggage that we carry right now and will be corrected with v1beta1 API). For example: `Image`, `Resources`
Would it make sense to move `RunAsRoot` under `Etcd.Spec.Etcd`?

Reply:
This is reflected in the pod spec (not separated by containers), i.e., I thought it makes more sense to _not_ move it under `.spec.etcd` (IIUC, `.spec.etcd` contains `etcd`-container specifics, while `.spec.backup` contains `backup-restore`-container specifics).

If you want, I can move it to `.spec.etcd` - I have no strong opinion here. Let me know!

Reply:
ok lets keep this at the `Etcd.Spec` level itself. In v1beta1 API we will move it.